website/source/docs/internals/plugins.html.md

```diff
@@ -0,0 +1,104 @@
+---
+layout: "docs"
+page_title: "Plugin System"
+sidebar_current: "docs-internals-plugins"
+description: |-
+  Learn about Vault's plugin system.
+---
+
+# Plugin System
+Certain Vault backends utilize plugins to extend their functionality outside of
+what is available in the core vault code. Often times these backends will
+provide both builtin plugins and a mechanism for executing external plugins.
+Builtin plugins are shipped with vault, often for commonly used implementations,
+and require no additional operator intervention to run. Builtin plugins are
+just like any other backend code inside vault. External plugins, on the other
+hand, are not shipped with the vault binary and must be registered to vault by
+a privileged vault user. This section of the documentation will describe the
+architecture and security of external plugins. 
+
+# Plugin Architecture
+Vault's plugins are completely separate, standalone applications that Vault
+executes and communicates with over RPC. This means the plugin process does not
+share the same memory space as Vault and therefore can only access the
+interfaces and arguments given to it. This also means a crash in a plugin can not
+crash the entirety of Vault.
+
+## Plugin Communication
+Vault creates a mutually authenticated TLS connection for communication with the
+plugin's RPC server. While invoking the plugin process Vault passes a [wrapping
+token](https://www.vaultproject.io/docs/concepts/response-wrapping.html) to the
+plugin process' environment. This token is single use and has a short TTL. Once
+unwrapped, it provides the plugin with a unique generated TLS certificate and
+private key for it to use to talk to the original vault process. 
+
+## Plugin Registration
+An important consideration of Vault's plugin system is to ensure the plugin
+invoked by vault is authentic and maintains integrity. There are two components
+that a Vault operator needs to configure before external plugins can be run.
```

Comment:
It's probably worth mentioning those two components in here: `There are two components that a Vault operator needs to configure before external plugins can be run: the plugin directory and the plugin's SHA256 sum.`